src/qtt/measurements/acquisition/uhfli_scope_reader.py

```diff
@@ -0,0 +1,322 @@
+
+        Returns:
+            The scope trigger slope (possible values are: Rise, Fall, Both or None).
+        """
+        return self.__uhfli.scope_trig_slope.get()
+
+    @trigger_slope.setter
+    def trigger_slope(self, slope: str) -> None:
+        """ Sets the external triggering slope.
+
+        Args:
+            slope: The external trigger slope (possible values are: Rise, Fall, Both and None).
+        """
+        self.__uhfli.scope_trig_slope.set(slope)
+
+    @property
+    def trigger_delay(self) -> float:
+        """ Gets the delay in seconds between the external trigger and acquisition.
+
+        Returns:
+            The scope trigger delay in seconds.
+        """
+        return self.__uhfli.scope_trig_delay.get()
+
+    @trigger_delay.setter
+    def trigger_delay(self, delay: float) -> None:
+        """ Sets the delay in seconds between the external trigger and acquisition.
+
+        Args:
+            delay: The scope trigger delay in seconds.
+        """
+        self.__uhfli.scope_trig_delay.set(delay)
+
+    @property
+    def enabled_channels(self) -> Tuple[int, ...]:
+        """ Gets the channel enabled states.
+
+        Returns:
+            The channels which are set to be enabled.
+        """
+        enabled_channels = self.__uhfli.scope_channels.get()
+        if enabled_channels == 3:
+            return 1, 2
+        return (enabled_channels, )
+
+    @enabled_channels.setter
+    def enabled_channels(self, value: Tuple[int, ...]):
+        """ Sets the given channels to enabled and turns off all others.
+
+        Args:
+            value: The channels which needs to be enabled.
+        """
+        self.__uhfli.scope_channels.set(sum(value))
+
+    def set_input_signal(self, channel: int, attribute: Optional[str]) -> None:
+        """ Adds an input channel to the scope.
+
+        Args:
+            channel: The input channel number.
+            attribute: The input signal to acquire.
+        """
+        channel_input = getattr(self.__uhfli, 'scope_channel{}_input'.format(channel))
+        channel_input.set(attribute)
+
+    def __get_uhfli_scope_records(self, number_of_records: int, timeout: float):
+        self.__uhfli.scope.execute()
+
+        records = 0
+        progress = 0
+        start = time.time()
+        while records < number_of_records or progress < 1.0:
+            records = self.__uhfli.scope.getInt("scopeModule/records")
+            progress = self.__uhfli.scope.progress()[0]
+            if time.time() - start > timeout:
+                error_text = "Got {} records after {} sec. - forcing stop.".format(records, timeout)
+                raise TimeoutError(error_text)
+
+        traces = self.__uhfli.scope.read(True)
+        wave_nodepath = '/{0}/scopes/0/wave'.format(self._address)
+        scope_traces = traces[wave_nodepath][:number_of_records]
+        return self.__covert_scope_data(scope_traces)
+
+    def __covert_scope_data(self, scope_traces):
```

Comment:
typo or are you covert-ing the scope_data?

Reply:
Yet another typo...